src/main/scala/org/constellation/infrastructure/endpoints/HealthCheckEndpoints.scala

```diff
@@ -0,0 +1,77 @@
+package org.constellation.infrastructure.endpoints
+
+import cats.effect.Concurrent
+import cats.syntax.all._
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import io.circe.syntax._
+import org.constellation.domain.healthcheck.HealthCheckConsensus.{FetchPeerHealthStatus, HealthConsensusCommand, NotifyAboutMissedConsensus, SendPerceivedHealthStatus}
+import org.constellation.domain.healthcheck.HealthCheckConsensusManager
+import org.constellation.domain.healthcheck.HealthCheckConsensusManager.SendProposalError
+import org.constellation.p2p.Cluster.EitherCodec._
+import org.http4s.dsl.Http4sDsl
+import org.http4s.HttpRoutes
+import org.http4s.circe._
+
+class HealthCheckEndpoints[F[_]](implicit F: Concurrent[F]) extends Http4sDsl[F] {
+  Slf4jLogger.getLogger[F]
+
+  def peerHealthCheckEndpoints(healthCheckConsensusManager: HealthCheckConsensusManager[F]) =
+    processConsensusCommand(healthCheckConsensusManager) <+>
+      getRoundProposal(healthCheckConsensusManager) <+>
+      getClusterState(healthCheckConsensusManager)
+
+  private def processConsensusCommand(
+    healthCheckConsensusManager: HealthCheckConsensusManager[F]
+  ): HttpRoutes[F] = HttpRoutes.of[F] {
+    case req @ POST -> Root / "health-check" / "consensus" =>
+      for {
+        cmd <- req.decodeJson[HealthConsensusCommand]
+        response <- cmd match {
+          case SendPerceivedHealthStatus(consensusHealthStatus, asProxyForId) =>
+            {asProxyForId match {
+              case Some(proxyToPeer) =>
+                healthCheckConsensusManager.proxySendHealthProposal(consensusHealthStatus, proxyToPeer)
+              case None =>
+                // are we sure that it will be processes no matter what and therefore we can return Accepted immediately?
+                F.start(healthCheckConsensusManager.handleConsensusHealthProposal(consensusHealthStatus))
+                  .map(_ => ().asRight[SendProposalError])
+            }}.flatMap(result => Ok(result.asJson))
+
+          case a @ NotifyAboutMissedConsensus(_, _, _) =>
+            F.start(healthCheckConsensusManager.handleNotificationAboutMissedConsensus(a))
+              .map(_ => ()) >>=
+              (result => Ok(result.asJson)) // I think we may need a separate endpoint here, as we are responding with different type, Unit here, Either above
```

Comment:
yes - imo you should unify the response. it should use a generic parameter based on the type of the command, so its response as well. then, it can return different payload within that scope.

Reply:
I wanted to avoid having broader Either[HealthCheckManagerError, Unit] when I will only be sending Unit, or responding with SendProposalError. Hence was thinking about a separate endpoint. The idea was to have the errors as precise as possible to match and handle precisely if needed. Not sure if changing SendProposalError to HealthCheckManagerError and having something like NotificationError extends HealthCheckManagerError is what you meant though?

Reply:
I meant:
```json
200 Ok
{
 "kind": "A",
 "data": { "name": "foo" }
}
```

```json
200 Ok
{
 "kind": "B",
 "data": { "age": 2 }
}
```

```json
400 Bad Request
{
 "message": "....",
}
```
or
```json
400 Bad Request
{
 "message": "....",
 "kind": "AbcError",
 "data": { "additionalMetadata": "", "round": "" }
}
```

Reply:
Ok I get the idea. That means a bit more work on the client and server side, but from the API point of view is probably more correct.